Automattic/simplenote-android
# Add Animated Actions

### Fix
Add animations to action icons shown in the app bar.  See the slowed down animations below for illustration.  The animations in the app are set to 400ms, which is equivalent to Android's standard [`config_mediumAnimTime`](https://developer.android.com/reference/android/R.integer#config_mediumAnimTime).

Action | Animation
-|-
Visibility On-to-Off | <a href="https://user-images.githubusercontent.com/3827611/75301844-31ab3b00-57f9-11ea-904d-ab749f4decba.gif"><img src="https://user-images.githubusercontent.com/3827611/75301844-31ab3b00-57f9-11ea-904d-ab749f4decba.gif" width="330"></a>
Visibility Off-to-On | <a href="https://user-images.githubusercontent.com/3827611/75301848-3374fe80-57f9-11ea-8dbb-5319a9b4cdea.gif"><img src="https://user-images.githubusercontent.com/3827611/75301848-3374fe80-57f9-11ea-8dbb-5319a9b4cdea.gif" width="330"></a>
Information Sheet | <a href="https://user-images.githubusercontent.com/3827611/75301362-d3ca2380-57f7-11ea-8649-169da4a750cc.gif"><img src="https://user-images.githubusercontent.com/3827611/75301362-d3ca2380-57f7-11ea-8649-169da4a750cc.gif" width="330"></a>
List Show-to-Hide | <a href="https://user-images.githubusercontent.com/3827611/75301855-3a9c0c80-57f9-11ea-9369-51d050237e92.gif"><img src="https://user-images.githubusercontent.com/3827611/75301855-3a9c0c80-57f9-11ea-9369-51d050237e92.gif" width="330"></a>
List Hide-to-Show | <a href="https://user-images.githubusercontent.com/3827611/75301856-3c65d000-57f9-11ea-83be-e281c712bd57.gif"><img src="https://user-images.githubusercontent.com/3827611/75301856-3c65d000-57f9-11ea-83be-e281c712bd57.gif" width="330"></a>
Toggle Checklist | <a href="https://user-images.githubusercontent.com/3827611/75301374-dc225e80-57f7-11ea-99b4-f4477e9caeb2.gif"><img src="https://user-images.githubusercontent.com/3827611/75301374-dc225e80-57f7-11ea-99b4-f4477e9caeb2.gif" width="330"></a>
Empty Trash | <a href="https://user-images.githubusercontent.com/3827611/75301379-e17fa900-57f7-11ea-9468-b51bdf187018.gif"><img src="https://user-images.githubusercontent.com/3827611/75301379-e17fa900-57f7-11ea-9468-b51bdf187018.gif" width="330"></a>

#### Note
All the animations have full opacity throughout the duration except for emptying the trash; it goes from 100% to 50% opacity since the disabled icon shown after the animation completes has 50% opacity.

The following animations can only be seen on a tablet-sized device since the associated icons are not shown in the app bar for phone-sized devices.
- Visibility On-to-Off
- Visibility Off-to-On
- List Show-to-Hide
- List Hide-to-Show

The `ic_visiblity_off_24dp` icon used when markdown is visible was also updated.  The strike across the icon was not centered.  Now it is.  See the screenshots below for illustration.

![av_visibility](https://user-images.githubusercontent.com/3827611/75301303-b2693780-57f7-11ea-913e-4d90a524978e.png)

### Test
Since all the icons, and therefore all the animations, are not shown on phone-sized devices, it's best to test on a tablet or both.
1. Tap any note in list with markdown enabled.
2. Long-press ***Hide List*** icon in app bar.
3. Notice ***Hide list*** message is shown.
4. Tap ***Hide List*** icon in app bar.
5. Notice list is hidden and ***Hide List*** icon animates to ***Show List*** icon.
6. Long-press ***Show List*** icon in app bar.
7. Notice ***Show list*** message is shown.
8. Tap ***Show List*** icon in app bar.
9. Notice list is shown and ***Show List*** icon animates to ***Hide List*** icon.
10. Long-press ***Visibility On*** icon in app bar.
11. Notice  ***Show Markdown*** message is shown.
12. Tap ***Visibility On*** icon in app bar.
13. Notice markdown is shown and ***Visibility On*** icon animates to ***Visibility Off*** icon.
14. Long-press ***Visibility Off*** icon in app bar.
15. Notice  ***Hide Markdown*** message is shown.
16. Tap ***Visibility Off*** icon in app bar.
17. Notice markdown is hidden and ***Visibility Off*** icon animates to ***Visibility On*** icon.
18. Long-press ***Toggle Checklist*** icon in app bar.
19. Notice  ***Toggle Checklist*** message is shown.
20. Tap ***Toggle Checklist*** icon in app bar.
21. Notice checkbox is added and ***Toggle Checklist*** icon animates.
22. Tap ***Toggle Checklist*** icon in app bar.
23. Notice checkbox is removed and ***Toggle Checklist*** icon animates.
24. Long-press ***Information*** icon in app bar.
25. Notice ***Information*** message is shown.
26. Tap ***Information*** icon in app bar.
27. Notice ***Information*** sheet is shown and ***Information*** icon animates.
28. Dismiss ***Information*** sheet.
29. Open navigation drawer.
30. Tap ***Trash*** item in navigation drawer.
31. If ***Trash*** is empty, notice disabled ***Trash*** icon in app bar.
32. If ***Trash*** is not empty, notice enabled ***Trash*** icon in app bar.
33. If ***Trash*** is empty, create and trash note.
34. Long-press ***Trash*** icon in app bar.
35. Notice ***Empty trash*** message is shown.
36. Tap ***Trash*** icon in app bar.
37. Notice ***Empty trash*** dialog is shown.
38. Tap ***No*** button in ***Empty trash*** dialog.
39. Notice ***Empty trash*** dialog is dismissed and ***Trash*** icon does not animate.
40. Tap ***Trash*** icon in app bar.
41. Notice ***Empty trash*** dialog is shown.
42. Tap ***Yes*** button in ***Empty trash*** dialog.
43. Notice ***Empty trash*** dialog is dismissed and ***Trash*** icon does animate.
44. Notice disabled ***Trash*** icon in app bar.

### Review
Only one developer and one designer are required to review these changes, but anyone can perform the review.

### Release
These changes do not require release notes.
